HOWTO_RELEASE.md

```diff
@@ -0,0 +1,106 @@
+# Steps
+
+Release early, release often. Don't be lazy.
+
+To use this doc: just replace 0.1.0 with the major.minor.patch version of
+the release. The sequence of commands below should be good to copy and
+paste, but do please pay attention to details!
+
+
+## Preparation
+
+- update main and run tests to verify all is green
+
+    git fetch upstream
+    git merge upstream/main
+    make tests
+
+
+### if it's a minor release (Z == 0)
+
+- tag `main` with only the minor version:
+
+    git tag 0.1
+    git push --tags
+
+- create a new release branch
+
+    git checkout -b release-0.1
+
+- create release notes after all main changes from last tag
+
+    git log --first-parent main --decorate > release-0.1.0.txt
+
+- tag the release (using those release notes)
+
+    git tag -s 0.1.0
+    git push --tags
+
+
+### if it's a micro release (Z != 0)
+
+- go to the release branch
+
+    git checkout release-0.1
+
+- cherry pick the needed commits from main:
+
+   git cherry-pick -m 1 COMMIT-HASH
+   ...
+
+- create release notes from the selected commits
+
+    git log
+
+- tag the release (using those release notes)
```

Comment:
How do I tag using release notes?

Reply:
As indicated in the next command (it will open you an editor).